cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -611,16 +613,34 @@ func (p *BicepProvider) Deploy(ctx context.Context) (*provisioning.DeployResult,
 		return nil, err
 	}
 
-	err = p.validatePreflight(
-		ctx,
-		bicepDeploymentData.Target,
-		bicepDeploymentData.CompiledBicep.RawArmTemplate,
-		bicepDeploymentData.CompiledBicep.Parameters,
-		deploymentTags,
-		optionsMap,
-	)
+	userConfig, err := p.userConfigManager.Load()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("loading user configuration: %w", err)
+	}
+
+	preflightDisableVar := "provision.disableValidation"
+	preflightDisable, exists := userConfig.GetString(preflightDisableVar)
```

Comment:
I think `userConfig.GetString()` can return `false` if the key exists but it is not a String.
https://github.com/vhvb1989/azure-dev/blob/main/cli/azd/pkg/config/config.go#L273